src/autopas/selectors/tuningStrategy/ActiveHarmony.h

```diff
@@ -0,0 +1,306 @@
+}
+
+bool ActiveHarmony::invalidateConfiguration() {
+  auto worstPerf = std::numeric_limits<double>::max();
+  addEvidence(worstPerf);
+  return true;
+}
+
+bool ActiveHarmony::tune(bool currentInvalid) {
+  if (currentInvalid) {
+    invalidateConfiguration();
+  }
+
+  // get configurations from server until valid newton3 option is found
+  do {
+    if (ah_fetch(htask) < 0) {
+      utils::ExceptionHandler::exception("ActiveHarmony::tune: Error fetching values from server");
+    }
+    fetchConfiguration();
+  } while (_allowedNewton3Options.find(_currentConfig.newton3) == _allowedNewton3Options.end() &&
+           invalidateConfiguration()); // short-circuit evaluation makes this only execute if newton3 option is invalid
+  AutoPasLog(debug, "ActiveHarmony::tune: Trying configuration {}.", _currentConfig.toString());
+
+  auto converged = ah_converged(htask);
+  if (converged) {
+    // set configuration to optimum
+    AutoPasLog(debug, "ActiveHarmony::tune: Reached converged state.");
+    if (ah_best(htask) != 0) {
+      utils::ExceptionHandler::exception("ActiveHarmony::tune: Error fetching best point.");
+    }
+    fetchConfiguration();
+    AutoPasLog(debug, "ActiveHarmony::tune: Selected optimal configuration {}.", _currentConfig.toString());
+  }
+  return !converged;
+}
+
+void ActiveHarmony::removeN3Option(Newton3Option option) {
+  _allowedNewton3Options.erase(option);
+  if (this->searchSpaceIsEmpty()) {
+    utils::ExceptionHandler::exception(
+            "ActiveHarmony::removeN3Option: Removing all configurations with Newton 3 {} caused the search space to be empty!",
+            option);
+  }
+}
+
+const Configuration &ActiveHarmony::getCurrentConfiguration() const {
+  return _currentConfig;
+}
+
+void ActiveHarmony::reset() {
+  // free memory
+  if (htask != nullptr) {
+    ah_leave(htask);
+    ah_kill(htask);
+  }
+  if (hdesc != nullptr) {
+    ah_close(hdesc);
+    ah_free(hdesc);
+  }
+  // initiate the tuning session
+  hdesc = ah_alloc();
+  if (hdesc == nullptr) {
+    utils::ExceptionHandler::exception("ActiveHarmony::reset: Error allocating Harmony descriptor.");
+  }
+
+  if (ah_connect(hdesc, nullptr, 0) != 0) {
+    utils::ExceptionHandler::exception("ActiveHarmony::reset: Error connecting to Harmony session.");
+  }
+  // set up tuning parameters
+  hdef_t *hdef = ah_def_alloc();
+  if (hdef == nullptr) {
+    utils::ExceptionHandler::exception("ActiveHarmony::reset: Error allocating definition descriptor");
+  }
+
+  if (ah_def_name(hdef, "AutoPas") != 0) {
+    utils::ExceptionHandler::exception("ActiveHarmony::reset: Error settings search name");
+  }
+
+  if (!_allowedCellSizeFactors->isFinite() || _allowedCellSizeFactors->size() > 1) {
+    if (ah_def_real(hdef, "cellSizeFactor", _allowedCellSizeFactors->getMin(), _allowedCellSizeFactors->getMax(),
+                    (_allowedCellSizeFactors->getMin(), _allowedCellSizeFactors->getMax()) / cellSizeSamples,
+                    nullptr) != 0) {
```

Comment:
So if I understand this correctly you always convert the passed cellSizeFactors to a sequence of 100 equidistant points between min and max. For the factors as a range, this makes sense if ActiveHarmony does not accept anything else but  I think you do not treat the finite case here?